libraries/AC_AttitudeControl/AC_PosControl.cpp

```diff
@@ -1102,16 +1100,10 @@ void AC_PosControl::get_stopping_point_xy_cm(Vector2p &stopping_point) const
     stopping_point = curr_pos.xy().topostype();
     float kP = _p_pos_xy.kP();
 
-    Vector3f curr_vel = _inav.get_velocity();
-
-    // if position controller is active remove the desired velocity component
-    if (is_active_xy()) {
-        curr_vel.x -= _vel_desired.x;
-        curr_vel.y -= _vel_desired.y;
-    }
+    Vector2f curr_vel = _inav.get_velocity().xy();
 
     // calculate current velocity
-    float vel_total = norm(curr_vel.x, curr_vel.y);
+    float vel_total = curr_vel.length();
```

Comment:
You're aware of course that this is a 3D length?

Reply:
curr_vel is a vector 2f or am I missing something?

Reply:
yes, you're right, sorry about that.